Jarvis-Platform/jarvis-studio
# Feature/typescript configuration

[comment]: [![Status](https://img.shields.io/badge/Status-DEVELOPMENT-orange.svg)]()
[comment]: [![Status](https://img.shields.io/badge/Status-HOLD-inactive.svg)]()

[![Status](https://img.shields.io/badge/Status-READY-brightgreen.svg)]()


## Description
Create a TypeScript configuration to start progressive migration.


## Todos
- [x] Add TypeScript packages

| Name   |   Version   | Type |
|--------|-----------|------|
| vue-class-component |  `^7.0.2` | `dependency` |
| vue-property-decorator |  `^8.3.0` | `dependency` |
| @vue/cli-plugin-typescript |  `^4.1.2` | `devDependency` |
| @vue/eslint-config-typescript |  `^4.0.0` | `devDependency` |
| typescript |  `~3.5.3` | `devDependency` |

- [x] Migrate `App.vue` file to class-component


## Deploy Notes
- Review code
- Try to run `serve` & `build` scripts
- Merge


## Steps to Test or Reproduce
Stop dev server, then:
```sh
git checkout master
git pull
git checkout feature/typescript-configuration
rm -rf node_modules/ && npm i
```
You can now start your devServer.
Everything should looks ok once server is running (no error).


## Impacted Areas in Application
New dependencies + `App.vue` & `main.ts`
